diff --git a/lua/noice/view/nui.lua b/lua/noice/view/nui.lua
index da9429e..57dc91c 100644
--- a/lua/noice/view/nui.lua
+++ b/lua/noice/view/nui.lua
@@ -68,12 +68,22 @@ function NuiView:update_options()
       if type(width) == "number" and type(height) == "number" then
         local col = self._opts.position and self._opts.position.col
         local row = self._opts.position and self._opts.position.row
+        if self._opts.border ~= "none" then
+          width = width + 2
+          height = height + 2
+        end
         self._opts.anchor = Util.nui.anchor(width, height)
         if self._opts.anchor:find("S") and row then
           self._opts.position.row = -row + 1
+          if self._opts.border ~= "none" then
+            self._opts.position.row = self._opts.position.row + 2
+          end
         end
         if self._opts.anchor:find("E") and col then
           self._opts.position.col = -col
+          if self._opts.border ~= "none" then
+            self._opts.position.col = col + 2
+          end
         end
       end
     else
